File: ADSO/Lenguajes/JavaScript/clases.js

class Player{
    // Atributos
    _nombre = ""
    _color = ""

    // Metodo Constructor
    constructor(nombre,color){
        this._nombre = nombre
        this._color = color
    }

    // Metodo Normal

    saludar(){
        return `hola mi nombre es ${this._nombre}, y soy de color ${this._color}`;
    }

    // Metodo Get y Set

    get nombre(){
        return this._nombre
    }

    set nombre(newNombre){
        this._nombre = newNombre
    }

    // Metodo Estatico

    static saludo(){
        return console.log("Hola pe")
    }

}

// Instanciar
let p1 = new Player("Juan","Rojo");
let p2 = new Player("Pedro","Azul");

console.log(p1.saludar())
console.log(p2.saludar())

// GET
console.log(p1.nombre);

// SET
p1.nombre = "Juancho"
console.log(p1.nombre);


// STATIC
Player.saludo()


//-------------------------------------------------------------------------//


// HERENCIA

class Animal {
    _name = ""

    set name(newName){
        this._name = newName
    }
}

class Perro extends Animal {
    _raza = ""

    set raza (newRaza){
        this._raza = newRaza
    }

    Presentacion(){
        return console.log(`Hola mi perr@ se llama ${this._name} y es de la raza ${this._raza}`)
    }
}



let perro = new Perro();

perro.name = "Kiara"
perro.raza = "Pug"

perro.Presentacion()

File: ADSO/Lenguajes/Ruby/clases.rb
class Player
  # Atributos
  attr_accessor :nombre, :color


  # Metodo Constructor

  def initialize(nombre = "Desconcido", color = "no seleccionado")
    @nombre = nombre
    @color = color
  end

  # Metodo Normal

  def saludar
    "hola mi nombre es #{@nombre}, y soy de color #{@color}"
  end

  # Metodo Get y Set

  def nombre
    @nombre
  end

=begin
  def initialize(nombre)
    @nombre = nombre
  end
=end

  # Metodo Estatico

  def self.saludo
    puts("Hola pe")
  end

end

# Instanciar
p1 = Player.new("Juan", "Rojo")
p2 = Player.new("Pedro", "Azul")

puts(p1.saludar)
puts(p2.saludar)

# GET
puts(p1.nombre)

# SET
p1.nombre = "Juancho"
puts(p1.nombre)

# STATIC
Player.saludo

#-------------------------------------------------------------------------

# HERENCIA

class Animal
  attr_accessor :name

  def initialize(name)
    @name = name
  end
end

class Perro < Animal
  attr_accessor :raza


  def initialize()
    @raza = raza
  end

  def Presentacion
    puts("Hola mi perr@ se llama #{@name} y es de la raza #{@raza}")
  end
end


perro = Perro.new()

perro.name = "Kiara"
perro.raza = "Pug"

perro.Presentacion()
